core/observables/observable.py

```diff
@@ -128,7 +128,12 @@ def add_text(cls, text, tags=[]):
             A saved Observable instance.
 
         """
-        o = Observable.guess_type(text).get_or_create(value=text)
+        if type is None:
+            guessed_type = Observable.guess_type(text)
+        else:
+            guessed_type = instantiate_subclass(Observable, type)
```

Comment:
As stated in my previous comment, please check for the return value of `instantiate_subclass` or catch an exception (otherwise, you will be calling `.get_or_create` on `None`, which will break)